syntax = "proto3";

package config.accounts;

import "google/protobuf/empty.proto";

service CurrentUserController {
    rpc GetCurrentUser(google.protobuf.Empty) returns (UserResponse) {}
    rpc UpdateCurrentUser(CurrentUserUpdateCurrentUserRequest) returns (UserResponse) {}
}

service CurrentUserProfileController {
    rpc GetCurrentUserProfile(google.protobuf.Empty) returns (UserProfileResponse) {}
    rpc UpdateCurrentUserProfile(CurrentUserProfileUpdateCurrentUserProfileRequest) returns (UserProfileResponse) {}
}

message CurrentUserProfileUpdateCurrentUserProfileRequest {
    string bio = 1;
    string avatar_url = 2;
}

message CurrentUserUpdateCurrentUserRequest {
    string email = 1;
    string first_name = 2;
    string last_name = 3;
}

message UserProfileResponse {
    optional int64 id = 1;
    optional UserResponse user = 2;
    optional string bio = 3;
    optional string avatar_url = 4;
    optional bool is_active = 5;
}

message UserResponse {
    optional int32 id = 1;
    string username = 2;
    optional string email = 3;
    optional string first_name = 4;
    optional string last_name = 5;
}

